relay-event-normalization/Cargo.toml

```diff
@@ -20,6 +20,7 @@ itertools = { workspace = true }
 maxminddb = { workspace = true }
 md5 = { workspace = true }
 once_cell = { workspace = true }
+psl = "2.1.33"
```

Comment:
We could source the [Public Suffix List](https://publicsuffix.org/list/public_suffix_list.dat) manually, but I assume you'd end up with a similar size. We could theoretically limit ourselves to the 100 most popular suffixes or something like that.

Reply:
Why is that link ~300kb but the list in the [psl crate](https://github.com/addr-rs/psl/blob/main/src/list.rs) 2.2mib?

Reply:
That's allegedly the same list compiled into rust code, but I cannot find the script that compiles it.

Reply:
It looks like the list is compiled using [psl_codegen](https://docs.rs/psl-codegen/latest/psl_codegen/fn.compile_psl.html) Maybe we can provide a shorter list of suffixes? That said, is the 2.2MiB size important? [crates.io](https://crates.io/crates/psl) says it's only 187KiB which seems alright, no?

Reply:
The size is the compressed size. If we need the crate (it's the best option) it's fine, but if there is a chance to not have an even bigger binary, then yeah I think it matters.

Reply:
That's fair!

I chose `psl` because it's the most popular on crates.io, and it's recommended by the Public Suffix List website for being fast (which, apparently, is at the cost of binary size). My thinking is that if someone's gone through the trouble of curating a full list of suffixes, we might as well use it!

I think the alternative would be to use the crates that `psl` itself relies on, and supply our own list of suffixes. This would (I assume) reduce the binary size (unclear how much) but then we'd have to maintain the suffix list ourselves, and would add some complexity to the repo.

I'll defer to your judgement on this, but it seems like using `psl`, if nothing else, has reasonable tradeoffs

Reply:
👍 thanks for the research, sounds good to me!